pkg/services/govmomi/vcenter/clone.go

```diff
@@ -49,8 +49,11 @@ func Clone(ctx *context.VMContext, bootstrapData []byte) error {
 	var extraConfig extra.Config
 	if len(bootstrapData) > 0 {
 		ctx.Logger.Info("applied bootstrap data to VM clone spec")
-		if err := extraConfig.SetCloudInitUserData(bootstrapData); err != nil {
-			return err
+		switch ctx.VSphereVM.Spec.ProvisionerType {
```

Comment:
This needs to be set in the `vspheremachine_controller` as it's responsible for generating the `VSphereVM` object

Reply:
I think it's already done on this line https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/5fe0b244d3ddfbe3a563bce9c7e187ef8fd957d3/controllers/vspheremachine_controller.go#L437

Reply:
oh right, I forgot that we were deepcopying, please discard my comment